apis/python/src/tiledbsoma/_exception.py

```diff
@@ -85,6 +85,51 @@ def is_already_exists_error(e: tiledb.TileDBError) -> bool:
     return "already exists" in stre.lower()
 
 
+class NotCreateableError(SOMAError):
+    """Raised when attempting to create an already existing SOMA object.
+
+    Lifecycle: experimental
+    """
+
+    pass
+
+
+def is_not_createable_error(e: tiledb.TileDBError) -> bool:
+    """Given a TileDBError, return true if it indicates the object cannot be created
+
+    Lifecycle: experimental
+
+    Example:
+        try:
+            tiledb.Array.create(uri, schema, ctx=ctx)
+                ...
+        except tiledb.TileDBError as e:
+            if is_not_createable_error(e):
+                ...
+            raise e
+    """
+    stre = str(e)
+    # Context:
+    # * A recurring paradigm in tiledbsoma.io is open for write (if exists) else create --
+    #   or, equivalently, create (if doesn't already exist), else open for write
+    # * A priori either seems fine
+    # * There are performance implications for trying the create first: when an
+    #   object _does_ already exist we get that quickly.
+    # * Therefore it's more performant to try-create-catch-open-for-write
+    # * However we have the following semantics for cloud URIs:
+    #   o For writes: must be "creation URIs" of the form "tiledb://namespace/s3://bucket/some/path"
+    #   o For read:   can be "creation URIs" _or_ non-creation URIs of the form
+    #     "tiledb://namespace/groupname" or "tiledb://namespace/uuid"
+    # * Put together: when we try-create-catch-open-for-write, _and when_ the URI provided
+    #   is a non-creation URI, we need to catch that fact and treat it as a non-error.
+    stre = stre.lower()
+    if "storage backend local not supported" in stre:
+        return True
+    if "storage backend not supported: local" in stre:
```

Comment:
Not seeing these strings in TileDB or TileDB-Py:
- https://github.com/search?q=repo%3ATileDB-Inc%2FTileDB+%22storage+backend+not+supported%22&type=code
- https://github.com/search?q=repo%3ATileDB-Inc%2FTileDB-Py%20storage%20backend%20not%20supported&type=code

Are they stable?

Reply:
@ryan-williams yes they are stable and no they aren't in those repos

Reply:
Happy to discuss Monday. And thank you! :)